sos/plugins/openstack_instack.py

```diff
@@ -28,6 +30,16 @@ def setup(self):
         self.add_copy_spec("/home/stack/instackenv.json")
         self.add_copy_spec("/home/stack/undercloud.conf")
 
+        # collect undercloud-debug: undercloud-debug.log will be created so we
+        # need to generate it in a tmp.dir, collect it now and then cleanup it
+        tmpdir = gettempdir()
```

Comment:
This is unsafe: as written, this is vulnerable to a `tmp` race. We spent a lot of effort to eliminate those from `sos` a few years ago: we really don't want to start introducing them again.

The problem is that this makes `/tmp/undercloud.debug` a predictable location that doesn't exist at the start, that will be created, and will then be collected. A malicious user just needs to symlink it to something else and wait for `sos` to access the symlink (either reading data it shouldn't, or writing to a location it shouldn't).

You need to call `Plugin.get_cmd_output_path()` (with `make=True` if this is the first command to be collected by the plugin), and then pass the returned string to the program (which would need a shell redirect in this case because undercloud-debug sucks). This uses `archive.get_tmp_dir()` and writes to the archive in-place so it is already within the private `/tmp` tree.

You would also need to use `call_ext_prog()` for that, with `runat=` the `sos_commands` dir - `get_cmd_output_now()` will not work if the program writes a named file (`/tmp/undercloud-debug.log`). It only works for programs that produce their output on stdout. Otherwise you would have to run the program and then wait for the file to appear and collect it with `add_copy_spec()` (like the old gluster debug stuff used to do).